Sources/BottomSheetView.swift

```diff
@@ -204,56 +213,73 @@ public final class BottomSheetView: UIView {
     }
 
     private func updateDimViewAlpha(for offset: CGFloat) {
-        if let superview = superview, let maxOffset = targetOffsets.last {
-            dimView.alpha = min(1, (superview.frame.height - offset) / (superview.frame.height - maxOffset))
+        if let superview = superview, let mainOffset = targetOffsets.first(where: { $0 < superview.frame.height }) {
+            dimView.alpha = min(1, (superview.frame.height - offset) / (superview.frame.height - mainOffset))
         }
     }
 
     // MARK: - UIPanGestureRecognizer
 
     @objc private func handlePan(panGesture: UIPanGestureRecognizer) {
-        guard let state = BottomSheetCalculator.translationState(
-            from: topConstraint.constant,
-            to: topConstraint.constant + panGesture.translation(in: superview).y,
-            targetOffsets: targetOffsets,
-            thresholds: targetThresholds,
-            currentTargetOffsetIndex: currentTargetOffsetIndex
-        ) else {
-            return
-        }
-
         switch panGesture.state {
         case .began:
             springAnimator.pauseAnimation()
+            initialOffset = topConstraint.constant
+
+        case .changed:
+            let translation = panGesture.translation(in: superview)
+            let location = initialOffset + translation.y
+
+            guard let model = models[location] else { return }
+
+            updateDimViewAlpha(for: location)
+            topConstraint.constant = model.nextOffset(for: location)
+
         case .ended, .cancelled, .failed:
-            animate(to: state.targetOffset)
-            if state.isDismissible {
+            let translation = panGesture.translation(in: superview)
+            let location = initialOffset + translation.y
+
+            guard let model = models[location] else { return }
```

Comment:
I find this subscript a bit confusing because it reads the same as default integer based subscript for indices. Could we perhaps just write it as `models.first({ $0.contains(offset: offset) })` in the beginnings of `handlePan` function?

Reply:
I agree